Create ui-c8y-1021-22-99-group-alarm-list-widget-does-not-include-sub-asset-alarms-after.md (#3487)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1021-22-99-group-alarm-list-widget-does-not-include-sub-asset-alarms-after.md

---
date: ""

> title: Alarm list widget does not include child devices alarms by default after upgrade to new implementation

Reviewer: ```suggestion
title: Alarm list widget now again shows child device alarms correctly
```

> title: Group alarm List widget does not include sub asset alarms after upgrade to new implementation (#9443) [GRAFT][release/y2025] (#9448)

Reviewer 2: ```suggestion
title: Alarm list widget does not include child devices alarms by default after upgrade to new implementation
```

product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-63974
version: 1021.22.99
---

> Co-authored-by: Lukasz Janusz <114157358+luja-c8y@users.noreply.github.com>
> Co-authored-by: Lukasz Janusz <lukasz.janusz@cumuloicty.com>

Reviewer 2: ```suggestion
In the migrated "Alarm list" widget, child device alarms were not being displayed even though they were shown by default in the legacy version. During migration, the toggle for showing child device alarms was incorrectly set to disabled. This issue has been fixed. The option now defaults to enabled, so migrated widgets will show child device alarms correctly without requiring configuration changes.
Note: The migration applies until the widget configuration is saved, then this change must be done manually.
```

> In the migrated "Alarm list" widget, child device alarms were not displayed although they were shown by default in the legacy version. During migration, the toggle for showing child device alarms was incorrectly set to disabled. This issue has been fixed. The option now defaults to enabled, so migrated widgets will show child device alarms correctly without requiring configuration changes.

Reviewer: @luja-c8y Which migration does this refer to? Shouldn´t we mention this?

Reviewer 2: @BeateRixen it refers to introducing a new version of alarm list widget. So "legacy" mean a alarm list widget based on angularJS with old design, and "migrated" means a widget based on an angular with a fresh designed.
In order to keep backwards compatibility we had to add some logic that converts legacy version of a widget to a new one. Original bug was introduced in this migration logic.
Maybe a "new version of an alarm widget" would better fit instead of "migration/migrated" like:

```
In the new version of "Alarm list" widget, child device alarms were not displayed although they were shown by default in the legacy version. During conversion to the new version of "Alarm list" widget, the toggle for showing child device alarms was incorrectly set to disabled. This issue has been fixed. The option now defaults to enabled, so converted widgets will show child device alarms correctly without requiring configuration changes.
```

Reviewer: @luja-c8y We could also just say "After migrating the "Alarms list" widget from Angular JS to Angular, ....?

Reviewer 2: @BeateRixen yes, I think in the end we could. I tried to avoid saying this clearly because redesign is visible, but used framework is not, but if this may explain it better then we could.

Reviewer: @janhommes @rahulc8y What is your take?

Reviewer 3: I would also rephrase it to new/old design.... migration is to technical and mostly the people don't care about angular to angularjs migrations.

> In the migrated "Alarm list" widget, child device alarms were not displayed although they were shown by default in the legacy version. During migration, the toggle for showing child device alarms was incorrectly set to disabled. This issue has been fixed. The option now defaults to enabled, so migrated widgets will show child device alarms correctly without requiring configuration changes.

Reviewer: ```suggestion
In a new design of the "Alarm list" widget, child device alarms were no longer displayed by default. This issue has been fixed. The new "Alarm list" widget now shows child device alarms correctly.
```

> In the migrated "Alarm list" widget, child device alarms were not being displayed even though they were shown by default in the legacy version. During migration, the toggle for showing child device alarms was incorrectly set to disabled. This issue has been fixed. The option now defaults to enabled, so migrated widgets will show child device alarms correctly without requiring configuration changes.

Reviewer: ```suggestion
In the migrated "Alarm list" widget, child device alarms were not displayed although they were shown by default in the legacy version. During migration, the toggle for showing child device alarms was incorrectly set to disabled. This issue has been fixed. The option now defaults to enabled, so migrated widgets will show child device alarms correctly without requiring configuration changes.
```

> Note: The migration applies until the widget configuration is saved, then this change must be done manually.

Reviewer: I don´t really get this sentence. Can you explain, please?

Reviewer 2: > Note: The migration applies until the widget configuration is saved, then this change must be done manually.

So, when a user creates an AngularJS version of a widget and then upgrades a C8Y UI version to one with a new implementation of the alarm widget written in Angular (and when dashboard with that widget is opened), the widget configuration conversion logic is triggered until the user opens and saves the widget configuration (even without a changes done in a form). Once this step has been completed, the old configuration will be overridden by the new one and the conversion logic will no longer be triggered.

Reviewer 3: I guess this is to technical and a detail which the customer might not need to know, or?

> Note: The migration applies until the widget configuration is saved, then this change must be done manually.

Reviewer: ```suggestion
```
